internal/setup/setup.go

```diff
@@ -772,3 +817,66 @@ func Select(release *Release, slices []SliceKey) (*Selection, error) {
 
 	return selection, nil
 }
+
+// pathInfoToYAML converts a PathInfo object to a yamlPath object.
+func pathInfoToYAML(pi *PathInfo) (*yamlPath, error) {
+	path := &yamlPath{
+		Mode:    yamlMode(pi.Mode),
+		Mutable: pi.Mutable,
+		Until:   pi.Until,
+		Arch:    yamlArch{List: pi.Arch},
+	}
+	switch pi.Kind {
+	case DirPath:
+		path.Dir = true
+	case CopyPath:
+		path.Copy = pi.Info
+	case TextPath:
+		path.Text = &pi.Info
+	case SymlinkPath:
+		path.Symlink = pi.Info
+	case GlobPath:
+	default:
+		return nil, fmt.Errorf("internal error: unrecognised PathInfo type: %s", pi.Kind)
+	}
+	return path, nil
+}
+
+// sliceToYAML converts a Slice object to a yamlSlice object.
+func sliceToYAML(s *Slice) (*yamlSlice, error) {
+	slice := &yamlSlice{
+		Essential: make([]string, 0, len(s.Essential)),
+		Contents:  make(map[string]*yamlPath, len(s.Contents)),
+		Mutate:    s.Scripts.Mutate,
+	}
+	for _, key := range s.Essential {
+		slice.Essential = append(slice.Essential, key.String())
+	}
+	sort.Strings(slice.Essential)
```

Comment:
Why are we touching the order here?  Isn't that coming from the slice definition on the release?  The human order will likely carry semantic that is more interesting than lexicographic order.

Reply:
Good point, discarded the change.